gap/attr.gi

```diff
@@ -1687,6 +1687,138 @@ function(D)
   return fail;
 end);
 
+InstallMethod(DigraphCore, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local N, lo, topdown, bottomup, hom, lo_var, image,
+  comps, comp, cores, G, vert_labels, D, in_core, n, m, L;
+  if not IsImmutableDigraph(digraph) then
+    digraph := DigraphImmutableCopy(digraph);
+  fi;
+  N := DigraphNrVertices(digraph);
+  if IsEmptyDigraph(digraph) then
+    if N >= 1 then
+      return [1];
+    else
+      return [];
+    fi;
+  fi;
+  digraph := ReducedDigraph(digraph);  # isolated verts are not in core
+  if DigraphHasLoops(digraph) then
+    return DigraphLoops(digraph){[1]};
+  elif not IsConnectedDigraph(digraph) then
+    comps := DigraphConnectedComponents(digraph).comps;
+    DigraphVertexLabels(digraph);
+    cores := [];
+    for comp in comps do
+      D := InducedSubdigraph(digraph, comp);
+      D := InducedSubdigraph(D,
+           Positions(BlistList(DigraphVertexLabels(D), DigraphCore(D)), true));
```

Comment:
I don't understand what this line is supposed to do.

Reply:
Thanks for your comments - I'd just been using `DigraphVertexLabels` as a way of retrieving prior vertex numbers, without really understanding how the vertices of a digraph can be labelled in whatever way we like.

This line creates the induced subdigraph of connected component `D` corresponding to its core - the reason it's not just `D := InducedSubdigraph(D, DigraphCore(D))` is to do with the vertex labelling. Since I've mistakenly been thinking of vertex labels and the list of vertices interchangeably, when I deal with the connected components here, `DigraphCore(D)` isn't necessarily a subset of the list of vertices of `D`, and so instead this line avoids running in to this problem by finding the positions of the elements of `DigraphCore(D)` within `DigraphVertexLabels(D)`. However, this will change after I've sorted out the vertex label misunderstanding in the rest of the function, and the line will probably end up much simpler.

Reply:
This'd be better if it was:
~~~~
D := InducedSubdigraph(D, DigraphVertexLabels(D){DigraphCore(D)};
~~~~
if that's really what you want to do.